sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -329,17 +339,21 @@ private[hive] class SparkExecuteStatementOperation(
     }
   }
 
-  override def cancel(): Unit = {
+  private def cancel(stateAfterCancel: OperationState): Unit = {
     synchronized {
       if (!getStatus.getState.isTerminal) {
         logInfo(s"Cancel query with $statementId")
-        setState(OperationState.CANCELED)
+        setState(stateAfterCancel)
         cleanup()
         HiveThriftServer2.eventManager.onStatementCanceled(statementId)
```

Comment:
One problem, if query timeout, we still call 
```
HiveThriftServer2.eventManager.onStatementCanceled(statementId)
```
then for timeout query, in spark thrift server ui, we still show it's cancelled,  we can't know this sql is timeout  directly.

IMO, We need to separate these two states.

Reply:
cc @juliuszsompolski  @wangyum

Reply:
Also support add a server side configuration about query TIMEOUT